Add documentation for setting up a Self-Hosted Endpoint (#75)
FriendlyCaptcha/friendly-docs: docs/guides/self-hosted-endpoint.md

All proxied requests need to send your proxy key as the value of the `X-Frc-Proxy-Key` header.

### 2. Configure your web server

Your web server needs to be configured to forward the widget requests to the Friendly Captcha API. These are the requests that must be forwarded:

```
GET  /api/v2/captcha/agent
GET  /api/v2/captcha/widget
POST /api/v2/captcha/activate
POST /api/v2/captcha/quote
POST /api/v2/captcha/redeem
```

For these requests, your server should forward the request in its entirety, including all headers. Additionally, the forwarded requests should include two ***extra*** headers:

1. `X-Frc-Proxy-Key`: The proxy key you generated in the Friendly Captcha dashboard.
2. `X-Frc-Proxy-Client-IP`: The original (source) IP address of the end user.

> As the upstream server for the proxied requests, you should specify

Reviewer: ```suggestion
Forward proxied requests to this endpoint (i.e., the upstream server):
```

```
global.proxy.frcapi.com
```

If you have access to [the EU Endpoint](./eu-endpoint.md), you may alternatively specify

```
eu.proxy.frcapi.com
```

[See below](#example-routing-configurations) for examples of how to configure a server to correctly proxy the widget requests.

:::warning Note

If you don't forward the entire request and its headers, the widget may still be functional, but the service will operate in a degraded state. If you forget either of the additional proxy headers (`X-Frc-Proxy-Key` and `X-Frc-Proxy-Client-IP`), the widget will display an error message. 

:::

### 3. Configure your site or app's widget

> The final step is to configure your server's URL as the API endpoint for your widget. This will ensure that the widget sends it request to your server (which will then forward them to the upstream Friendly Captcha API). If your server's URL is `https://organization.org` and you configure the widget using HTML `data-` attributes, your markup will look something like this:

Reviewer: ```suggestion
The final step is to configure your server's URL as the API endpoint for your widget. This will ensure that the widget sends its requests to your server (which will then forward them to the upstream Friendly Captcha API). If your server's URL is `https://organization.org` and you configure the widget using HTML `data-` attributes, your markup will look something like this:
```

# Self-Hosted Endpoint

:::info

The Self-Hosted Endpoint feature is only available for Friendly Captcha v2.

:::

When a customer website or application loads a Friendly Captcha widget, the widget makes a number of requests to the Friendly Captcha API. The API endpoint is `global.frcapi.com`, or `eu.frcapi.com` for customers who use [the EU Endpoint](./eu-endpoint). Friendly Captcha offers the **Self-Hosted Endpoint** feature for customers who prefer to have all end-user traffic funneled through their own infrastructure.

To use a Self-Hosted Endpoint, customers operate a proxy server that forwards widget requests to the Friendly Captcha API. Because the Friendly Captcha service depends on data sent by the widget, care must be taken when configuring Self-Hosted Endpoints to ensure that the data is correctly forwarded.

## Setup

There are 3 steps to setting up a Self-Hosted Endpoint.

1. Generate a proxy key in the Friendly Captcha dashboard.
2. Configure your web server to proxy widget requests to the Friendly Captcha API.

> 3. Configure your widget to use your Self-Hosted Endpoint.

Reviewer: I would maybe remove this chunk, or replace it with just `To set up a Self-Hosted Endpoint, follow the 3 steps below:` or similar.

Alternatively, it could be like a mini table of contents for the steps (where each bullet matches the wording of each heading and links to it).

In its current form, it's easy to miss that the bullet points match with the three steps below it.
